logs_ingest/filtering.py

```diff
@@ -0,0 +1,130 @@
+#   Copyright 2021 Dynatrace LLC
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+import fnmatch
+import os
+import re
+from typing import Dict, Set
+
+from logs_ingest import logging
+from logs_ingest.mapping import severity_to_log_level_dict, log_level_to_severity_dict, RESOURCE_TYPE_ATTRIBUTE, \
+    RESOURCE_ID_ATTRIBUTE
+
+GLOBAL = "global"
+FILTER_NAMES_PREFIXES = ["FILTER.RESOURCE_TYPE.MIN_LOG_LEVEL.", "FILTER.RESOURCE_TYPE.CONTAINS_PATTERN.",
+                         "FILTER.RESOURCE_ID.MIN_LOG_LEVEL.","FILTER.RESOURCE_ID.CONTAINS_PATTERN."]
+
+
+class LogFilter:
+    def __init__(self):
+        self._filter_config: str = os.environ.get("FILTER_CONFIG", "")
+        logging.info(f"Filter_config: {self._filter_config}")
+        filter_config_pattern = re.compile(r'([^;\s].+?)=([^;]*)')
+        self._filters_tuples = filter_config_pattern.findall(self._filter_config)
+        self._filters_tuples = [modified_filter_tuple for filter_tuple in self._filters_tuples
+                                if (modified_filter_tuple := self._prepare_filters_tuples(filter_tuple)) is not None]
+        self.filters_dict = self._prepare_filters_dict()
+        if self._filter_config and not self.filters_dict:
+            logging.info("Parsing filter config failed.")
+
+    @staticmethod
+    def _prepare_filters_tuples(filter_tuple):
+        filter_name = filter_tuple[0].strip().casefold()
+        value = filter_tuple[1].strip()
+        if filter_name.startswith("FILTER.GLOBAL.MIN_LOG_LEVEL".casefold()):
+            key = GLOBAL
+            return key, (filter_name, value)
+
+        if filter_name.startswith("FILTER.GLOBAL.CONTAINS_PATTERN".casefold()):
+            key = GLOBAL
+            return key, (filter_name, value)
+
+        for prefix in FILTER_NAMES_PREFIXES:
+            prefix = prefix.casefold()
+            if filter_name.startswith(prefix):
+                key = filter_name.split(prefix)[1].casefold()
+                if key:
+                    return key, (filter_name, value)
+
+        return None
+
+    def _prepare_filters_dict(self) -> Dict:
+        grouped_filters = self._group_filters()
+        filters_to_apply_dict = {}
+        for k, filter_name_value_dict in grouped_filters.items():
+            filters_to_apply = []
+            for filter_name, filter_value in filter_name_value_dict.items():
+                if "MIN_LOG_LEVEL".casefold() in filter_name:
```

Comment:
you can replace casefold call with simple `min_log_level`. The same applies to other places literals are used, like in `44`, `48` and `72`